src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-wizard/rgw-multisite-wizard.component.ts

```diff
@@ -86,64 +105,78 @@ export class RgwMultisiteWizardComponent extends BaseModal implements OnInit {
-              map((daemonStats) => ({
-                hostname: daemon.server_hostname,
-                port: daemon.port,
-                frontendConfig: daemonStats['rgw_metadata']['frontend_config#0']
-              }))
-            )
-          );
-          return forkJoin(daemonStatsObservables);
-        })
-      )
-      .subscribe((daemonStatsArray) => {
-        this.rgwEndpoints.value = daemonStatsArray.map((daemonStats) => {
-          const protocol = daemonStats.frontendConfig.includes('ssl_port') ? 'https' : 'http';
-          return `${protocol}://${daemonStats.hostname}:${daemonStats.port}`;
-        });
-        const options: SelectOption[] = this.rgwEndpoints.value.map(
-          (endpoint: string) => new SelectOption(false, endpoint, '')
-        );
-        this.rgwEndpoints.options = [...options];
-      });
-
+    this.open = this.route.outlet === 'modal';  
+    this.loadRGWEndpoints();
     this.multiClusterService.getCluster().subscribe((clusters) => {
-      this.clusterDetailsArray = Object.values(clusters['config'])
-        .flat()
-        .filter((cluster) => cluster['url'] !== clusters['current_url']);
+      const currentUrl = clusters['current_url'];
+      this.clusterDetailsArray = Object.values(clusters['config']).flat()
+        .filter((cluster) => cluster['url'] !== currentUrl);
       this.isMultiClusterConfigured = this.clusterDetailsArray.length > 0;
-      if (!this.isMultiClusterConfigured) {
-        this.stepTitles = STEP_TITLES_SINGLE_CLUSTER.map((title) => ({
-          label: title
+      this.stepTitles = (this.isMultiClusterConfigured ? STEP_TITLES_MULTI_CLUSTER_CONFIGURED : STEP_TITLES_SINGLE_CLUSTER)
+        .map((label, index) => ({
+          label,
+          onClick: () => (this.currentStep.stepIndex = index)
         }));
-        this.stepTitles.forEach((steps, index) => {
-          steps.onClick = () => (this.currentStep.stepIndex = index);
-        });
-      } else {
-        this.selectedCluster = this.clusterDetailsArray[0]['name'];
-      }
       this.wizardStepsService.setTotalSteps(this.stepTitles.length);
+      this.selectedCluster = this.isMultiClusterConfigured ? this.clusterDetailsArray[0]['name'] : null;
     });
-
+  
     this.summaryService.subscribe((summary) => {
-      this.executingTask = summary.executing_tasks.filter((tasks) =>
-        tasks.name.includes('progress/Multisite-Setup')
-      )[0];
+      this.executingTask = summary.executing_tasks.find(task =>
+        task.name.includes('progress/Multisite-Setup')
+      );
+    });
+  
+    this.stepTitles.forEach((step) => {
+      this.stepsToSkip[step.label] = false;
+    });
+  
+    this.rgwRealmService.getRealmTokens().subscribe((data: { realm: string; token: string }[]) => {
+      const base64Matcher = /^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{4})$/;
+      this.realmsInfo = data.filter(realmInfo => base64Matcher.test(realmInfo.token));
```

Comment:
Why do we need to do this ? We are always setting the first value that comes out of it. Can this happen that tokens are not encoded in base64 ? and if there are no base64 tokens what do we show in UI ? Just curious

Reply:
We need to validate if it is a valid token because in some cases such as :
1. Endpoints are not configured on the master zone we get - `Endpoints not configured` in the token
2. access/secret keys are not configured on master zone we get - `access/secret keys not defined`
So to avoid selecting realms with this content in the token, we need validation